lib/std/error_set.zig

```diff
@@ -0,0 +1,147 @@
+/// error_to_exclude does not have to be a member of BaseErrorSet.
+/// See also `WithoutAssertContained`.
+pub fn Without(comptime BaseErrorSet: type, comptime error_to_exclude: anyerror) type {
+    const ErrorSetToExclude = @TypeOf(@field(anyerror, @errorName(error_to_exclude)));
+    return comptime Excluding(BaseErrorSet, ErrorSetToExclude);
+}
+test Without {
+    comptime {
+        assert(Without(error{ A, B, C }, error.B) == error{ A, C });
+        assert(Without(error{ A, B }, error.A) == error{B});
+        assert(Without(error{ A, B }, error.C) == error{ A, B });
+        assert(Without(error{D}, error.D) == error{});
+        assert(Without(error{}, error.A) == error{});
+    }
+}
+
+/// Returns an error set with all errors which are
+/// members of both ErrorSetA and ErrorSetB.
+/// The resulting error set may be empty.
+/// See also `IntersectAssertNonEmpty`.
+pub fn Intersect(comptime ErrorSetA: type, comptime ErrorSetB: type) type {
+    if (ErrorSetA == anyerror) return ErrorSetB; //allows intersecting with anyerror
+
+    const base_info = @typeInfo(ErrorSetA);
+    comptime assert(base_info == .ErrorSet);
+    const non_anyerror_base_info = base_info.ErrorSet.?; //anyerror checked above
+    comptime var remaining_error_count = 0;
+    comptime var remaining_error_buffer: [non_anyerror_base_info.len]Type.Error = undefined;
+    inline for (non_anyerror_base_info) |error_of_set| {
+        if (comptime contains(ErrorSetB, @field(anyerror, error_of_set.name))) {
+            remaining_error_buffer[remaining_error_count] = error_of_set;
+            remaining_error_count += 1;
+        }
+    }
+    return @Type(.{ .ErrorSet = remaining_error_buffer[0..remaining_error_count] });
+}
+test Intersect {
+    comptime {
+        assert(Intersect(error{ A, B }, error{ B, D }) == error{B});
+        assert(Intersect(error{ B, D }, error{ B, C, D, E }) == error{ B, D });
+        assert(Intersect(error{ A, E, F }, anyerror) == error{ A, E, F });
+        assert(Intersect(anyerror, anyerror) == anyerror);
+        assert(Intersect(error{}, anyerror) == error{});
+        assert(Intersect(error{ A, B }, error{G}) == error{});
+        assert(Intersect(error{}, error{}) == error{});
+    }
+}
+
+/// Returns the set of errors that are
+/// members of BaseErrorSet, but not members of ToExcludeErrorSet.
+/// Asserts that all members of ToExcludeErrorSet are members of BaseErrorSet.
+/// See also `Excluding`.
+pub fn ExcludingAssertAllContained(comptime BaseErrorSet: type, comptime ToExcludeErrorSet: type) type {
+    comptime assert(containsAll(BaseErrorSet, ToExcludeErrorSet));
+    return comptime Excluding(BaseErrorSet, ToExcludeErrorSet);
+}
+/// Returns an error set with all members of BaseErrorSet
+/// except for error_to_exclude.
+/// Asserts that error_to_exclude is a member of BaseErrorSet.
+/// See also `Without`.
+pub fn WithoutAssertContained(comptime BaseErrorSet: type, comptime error_to_exclude: anytype) type {
```

Comment:
Oversight: `error_to_exclude` should be `: anyerror` here, but I don't want to re-bump the CI just for this.

Reply:
Made that change and also added a commit that uses `std.error_set` for `SelfExePathError`